configure.ac

```diff
@@ -1216,53 +1216,47 @@ dnl it would break GCC's #include_next.
 AC_DEFUN([SUPPRESS_WARNINGS],
          [$(echo $1 |${SED} -E -e 's/(^| )-I/\1-isystem /g' -e 's;-isystem /usr/include([/ ]|$);-I/usr/include\1;g')])
 
-dnl enable-fuzz should disable all other targets
```

Comment:
On a second thought, this might be useful to keep. We don't want to waste resources compiling the wallet, util, cli, qt, bench, ... when the goal is to only fuzz.

Reply:
Sorry, I'm confused by this. enable-fuzz refers to whether the fuzz targets should be built, which is now being done by default. It seems that we only want to disable the other targets when building with libfuzzer, afl or honggfuzz (the latter two of which still need to be addressed, which should be easy enough by inspecting CXX_FLAGS). Does this sound right to you?

Reply:
Before your change enable-fuzz means build the fuzz targets, but disable all other targets
After your change, it should keep that meaning, so that we don't waste compile resources. (I incorrectly assumed that it should go or change meaning).

After your change, the only thing that should change is that the fuzz target is always built (whether enable-fuzz or not).

There *could* be a disable-fuzz option that disables the fuzz target, but that can be done in a follow up. Unless you want to do it here right away.

Reply:
@MarcoFalke I think we're giving @danben conflicting instructions here?

Did you see https://github.com/bitcoin/bitcoin/pull/20936#issuecomment-766629435. I don't think we can just keep --{enable,disable}-fuzz with the same meaning.

Reply:
I'm working on a patch that implement's @MarcoFalke's instructions as best I understand them just so that no matter how this gets resolved there will be a good place to begin wrapping things up.

The one thing that seemed a bit odd to me is this: if enable-fuzz used to mean "build the fuzz targets, but disable all other targets" and the expected change is that fuzz targets get built regardless of the value of enable-fuzz, then we end up with a flag whose name is enable-fuzz but all it does is disable building of non-fuzz targets.

Additionally, under these assumptions I don't think it's possible to punt on the flag that disables fuzz targets because without the ability to change code, some of the CI tests won't pass. But unfortunately this flag cannot be called disable-fuzz if we are saying that enable-fuzz has no effect on whether fuzz targets are built.

Reply:
In my comment I assumed that enable-fuzz and disable-fuzz are separate concepts (I don't know if configure can treat them as such). I used different words to describe the same that @sipa did in the previous comment. I don't care about the names of the configure arguments, as long as they achieve what we want :)